interfaces/udev/backend.go

```diff
@@ -95,8 +103,46 @@ func (b *Backend) Setup(appSet *interfaces.SnapAppSet, opts interfaces.Confineme
 			// FIXME: somehow detect the interfaces that were
 			// disconnected and set subsystemTriggers appropriately.
 			// ATM, it is always going to be empty on disconnect.
-			return b.reloadRules(subsystemTriggers)
+			if err := b.reloadRules(subsystemTriggers); err != nil {
+				return err
+			}
+		}
+	}
+
+	if udevSpec.ControlsDeviceCgroup() {
+		// The spec states that the snap can manage its own device
+		// cgroup (typically applies to container-like snaps), in which
+		// case leave a flag for snap-confine in at a known location.
+
+		if err := os.MkdirAll(dirs.SnapCgroupPolicyDir, 0755); err != nil {
+			return fmt.Errorf("cannot create directory for cgroup flags: %w", err)
+		}
+
+		var buf bytes.Buffer
+		buf.WriteString("# This file is automatically generated.\n")
+		buf.WriteString("# snap is allowed to manage own device cgroup.\n")
+		buf.WriteString("self-managed=true\n")
+
+		err := osutil.EnsureFileState(selfManageDeviceCgroupPath, &osutil.MemoryFileState{
```

Comment:
The file is not cleaned up on snap remove, I'll include that in the next batch of patches.